### PR TITLE
improve performance by turning off CP state tracking

### PR DESCRIPTION
**Pull request by adon-at-work**
- turn off the CP state tracking

### CODE REVIEW

**Comment by adon-at-work** [comment]
@neraliu , please check

File: src/context-parser-handlebars.js [hunk 1]
```diff
@@ -392,7 +392,7 @@ ContextParserHandlebars.prototype.analyzeAst = function(ast, contextParser, char
                 node.type === handlebarsUtils.RAW_EXPRESSION) {
 
                 // lookupStateForHandlebarsOpenBraceChar from current state before handle it
-                parser.setCurrentState(ContextParserHandlebars.lookupStateForHandlebarsOpenBraceChar[parser.state]);
+                parser.state = ContextParserHandlebars.lookupStateForHandlebarsOpenBraceChar[parser.state];
```
Comment (adon-at-work) [inline src/context-parser-handlebars.js]: this particular line of change is unncessary if https://github.com/yahoo/context-parser/pull/39 is merged

Comment (neraliu) [inline src/context-parser-handlebars.js]: should we use `reset()` api?

Comment (adon-at-work) [inline src/context-parser-handlebars.js]: i believe No. this particular line of code is meant to make the contextual analyzer work properly with a template as if the handlebars syntaxes are missing. you still remember what you wrote? :)

**Comment by adon-at-work** [comment]
rebased according to the new master to resolve conflict
@neraliu

File: src/context-parser-handlebars.js (hunk 1 above)
Comment (neraliu) [inline src/context-parser-handlebars.js]: got it. but why do we not using the api `setCurrentState()`? instead of assigning directly to the internal var?

Comment (adon-at-work) [inline src/context-parser-handlebars.js]: yes we should. but the changes applied in yahoo/context-parser#39 is not released yet

Comment (neraliu) [inline src/context-parser-handlebars.js]: let's wait for the CP v2.0.2 release and use the api for this PR.

**Pull request merged**